k9mail/src/main/java/com/fsck/k9/activity/RecipientLoader.java

```diff
@@ -0,0 +1,300 @@
+package com.fsck.k9.activity;
+
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import android.annotation.TargetApi;
+import android.content.AsyncTaskLoader;
+import android.content.Context;
+import android.database.Cursor;
+import android.net.Uri;
+import android.os.Build.VERSION_CODES;
+import android.os.CancellationSignal;
+import android.os.OperationCanceledException;
+import android.provider.ContactsContract;
+
+import com.fsck.k9.mail.Address;
+import com.fsck.k9.view.RecipientSelectView.Recipient;
+import com.fsck.k9.view.RecipientSelectView.RecipientCryptoStatus;
+
+
+@TargetApi(VERSION_CODES.JELLY_BEAN) // TODO get rid of this, affects cancellation behavior!
+public class RecipientLoader extends AsyncTaskLoader<List<Recipient>> {
```

Comment:
`Loader` is a mechanism that helps doing work off the main thread. The actual work should be performed by a dedicated class that is not tied to the mechanism.

I created issue #965 to change this later.